src/shared/firewall-util-nft.c

```diff
@@ -941,6 +944,105 @@ int nft_set_element_modify_iprange(
         return sd_nfnl_call_batch(ctx->nfnl, &m, 1, NFNL_DEFAULT_TIMEOUT_USECS, NULL);
 }
 
+static int nft_message_append_setelem_ip4(
```

Comment:
Not a matter, but I do not think this and `_ip6` should be split out from `nft_set_element_modify_ip()`. We can merge the three function like the follow:
```c
int nft_set_element_modify_ip(...) {
        ...
        if (!source)
                return -EINVAL;

        r = sd_nfnl_nft_message_new_setelems(ctx->nfnl, &m, add, nfproto, table, set);
        if (r < 0)
                return r;

        r = sd_netlink_message_open_container(m, NFTA_SET_ELEM_LIST_ELEMENTS);
        if (r < 0)
                return r;

        r = sd_nfnl_nft_message_append_setelem(m, 0, &source, FAMILY_ADDRESS_SIZE(af), NULL, 0, 0);
        if (r < 0)
                return r;

        r = sd_netlink_message_close_container(m); /* NFTA_SET_ELEM_LIST_ELEMENTS */
        if (r < 0)
                return r;

        return sd_nfnl_call_batch(ctx->nfnl, &m, 1, NFNL_DEFAULT_TIMEOUT_USECS, NULL);
}
```

Reply:
nice!

Reply:
done